addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/AbstractNetatmoThingHandler.java

```diff
@@ -83,9 +83,11 @@ protected void updateChannels() {
         getThing().getChannels().stream().filter(channel -> channel.getKind() != ChannelKind.TRIGGER)
                 .forEach(channel -> {
                     String channelId = channel.getUID().getId();
-                    State state = getNAThingProperty(channelId);
-                    if (state != null) {
-                        updateState(channel.getUID(), state);
+                    if (isLinked(channelId)) {
```

Comment:
If we are using filters anyway I would prefer this as an extra filter.

@htreu should we always or never do this, or just like @lolodomo now decided because it makes sure we are not downloading big things in which we are not interested.

Reply:
I don't see how this check prevents the binding from _not_ downloading anything. The #getNAThingProperty calls only operate on already present data.
Since the #stateUpdate call will already filter unlinked channels I consider this not to be necessary.

Reply:
Not exactly, getNAThingProperty will call HttpUtil.downloadImage with the already present URL data to update the image channel. This is useless and unexpected in case the channel is uinlinked.
Adding this test, you are sure to build and update state channel only for channels that are used by the user.

Reply:
ah, I see in the `NAWelcomeCameraHandler`. wouldn't it make sense to add the check right there then?

Reply:
Now I see measurableChannels which is storing linked channels. But it is not used by getNAThingProperty of the parent class.
One alternative to my change would be to move the check "isLinked" to each getNAThingProperty.

Reply:
I should do it in all handlers, not only in NAWelcomeCameraHandler.
The advantage of my current code is that this is done only at one place for all handlers.

Reply:
Note that getNAThingProperty on the abstract class will return UNDEF if the channel is not linked (not inside measurableChannels). So the test of the state with null looks wrong.
getNAThingProperty should probably return null if the channel is not linked ?

Reply:
I think we should not update all getNAThingProperty and just do the test like I propose.

Reply:
After all I guess its a matter of taste. Its okay to check the link status of a channel prior to an update, so nothing wrong here.